pages/recipes/[id]/index.js

```diff
@@ -6,13 +8,27 @@ import {
   StyledDetailsPageContainer,
   StyledDetailsItem,
   StyledItemsRow,
+  EditDeleteDiv,
 } from "@/components/StyledDetailsPage.js";
+import {
+  StyledButtonNo,
+  StyledButtonYes,
+  StyledDeleteButton,
+} from "@/components/Modal/ModalStyle.js";
 
 export default function DetailsPage() {
+  const [showModal, setShowModal] = useState(false);
   const router = useRouter();
   const { isReady } = router;
   const { id } = router.query;
   const { data: recipe, isLoading, error } = useSWR(`/api/recipes/${id}`);
+  async function deleteRecipe() {
+    await fetch(`/api/recipes/${id}`, {
+      method: "DELETE",
```

Comment:
You don't seem to be checking for this request method in your API route. 

Also, i don't see where you are deleting the item from the database.